lib/geocoder/calculations.rb

```diff
@@ -38,12 +38,7 @@ module Calculations
     # Returns true if all given arguments are valid latitude/longitude values.
     #
     def coordinates_present?(*args)
-      args.each do |a|
-        # note that Float::NAN != Float::NAN
-        # still, this could probably be improved:
-        return false if (!a.is_a?(Numeric) or a.to_s == "NaN")
-      end
-      true
+      args.map { |a| (a.is_a? Numeric and !a.to_f.nan?) ? true : false }.all?
```

Comment:
Just a drive-by suggestion for simplifying this further:
  - The expression is already a boolean, the ternary `? true : false` is not needed
  - [`Enumerable#all?`](https://ruby-doc.org/core-2.3.1/Enumerable.html#method-i-all-3F) takes a block argument:

```ruby
args.all? { |a| a.is_a? Numeric and !a.to_f.nan? }
```